mne/channels/interpolation.py

```diff
@@ -251,25 +273,82 @@ def _interpolate_bads_nirs(inst, method="nearest", exclude=(), verbose=None):
     chs = [inst.info["chs"][i] for i in picks_nirs]
     locs3d = np.array([ch["loc"][:3] for ch in chs])
 
-    _check_option("fnirs_method", method, ["nearest", "nan"])
-
-    if method == "nearest":
-        dist = pdist(locs3d)
-        dist = squareform(dist)
-
-        for bad in picks_bad:
-            dists_to_bad = dist[bad]
-            # Ignore distances to self
-            dists_to_bad[dists_to_bad == 0] = np.inf
-            # Ignore distances to other bad channels
-            dists_to_bad[bads_mask] = np.inf
-            # Find closest remaining channels for same frequency
-            closest_idx = np.argmin(dists_to_bad) + (bad % 2)
-            inst._data[bad] = inst._data[closest_idx]
-    else:
-        assert method == "nan"
-        inst._data[picks_bad] = np.nan
+    dist = pdist(locs3d)
+    dist = squareform(dist)
+
+    for bad in picks_bad:
+        dists_to_bad = dist[bad]
+        # Ignore distances to self
+        dists_to_bad[dists_to_bad == 0] = np.inf
+        # Ignore distances to other bad channels
+        dists_to_bad[bads_mask] = np.inf
+        # Find closest remaining channels for same frequency
+        closest_idx = np.argmin(dists_to_bad) + (bad % 2)
+        inst._data[bad] = inst._data[closest_idx]
+
     # TODO: this seems like a bug because it does not respect reset_bads
     inst.info["bads"] = [ch for ch in inst.info["bads"] if ch in exclude]
 
     return inst
+
+
+@verbose
+def _interpolate_bads_seeg(inst, exclude=None, tol=2e-3, verbose=None):
+    if exclude is None:
+        exclude = list()
+    picks = pick_types(inst.info, meg=False, seeg=True, exclude=exclude)
+    inst.info._check_consistency()
+    bads_idx = np.isin(np.array(inst.ch_names)[picks], inst.info["bads"])
+
+    if len(picks) == 0 or bads_idx.sum() == 0:
+        return
+
+    pos = inst._get_channel_positions(picks)
+
+    # Make sure only sEEG are used
+    bads_idx_pos = bads_idx[picks]
+
+    # for each bad contact:
+    # 1) find nearest neighbor to define the electrode shaft line
+    # 2) find all contacts on the same line
+    # 3) interpolate the bad contacts
+
+    dist = squareform(pdist(pos))
+    np.fill_diagonal(dist, np.inf)
+
+    picks_bad = list(np.where(bads_idx_pos)[0])
+    while picks_bad:
```

Comment:
can we avoid while loops and use a plain for? I find them usually unecessary and error prone.

Reply:
The while loop was used because you don't know if two bad channels are on the same shaft until you find the shaft. I can't think of an easy way around that with a for loop.

Reply:
I wonder if a helper function (`_find_shafts` or so) that returns a list-of-lists of channel indices would be generally useful? If so, it could be used here to pre-compute the channels into shaft groupings, then a for-loop over `picks_bad` would work.

Reply:
Sure, we could do that. I was just finding them for the bad contacts to save the computation but it's a very small amount

Reply:
Hmm, I wrote that but even if you know the shafts, you still have to iterate over `picks_bad` (list of bad indices) and you don't know whether two are on the same shaft or not so you have to use a while loop.